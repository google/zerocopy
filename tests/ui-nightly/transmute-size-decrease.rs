// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
extern crate zerocopy;

// It is unsupported to decrease the size of a transmuted value.
const SIZE_DECREASE: [u8; 1] = zerocopy::transmute([0u8; 2]);

fn main() {
    _ = SIZE_DECREASE;
}
