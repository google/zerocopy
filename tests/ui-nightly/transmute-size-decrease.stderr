error[E0080]: evaluation of `<[u8; 1] as zerocopy::TransmuteFrom<[u8; 2]>>::IS_TRANSMUTABLE` failed
 --> src/lib.rs
  |
  |         assert!(mem::size_of::<Src>() == mem::size_of::<Dst>());
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: mem::size_of::<Src>() == mem::size_of::<Dst>()', $DIR/src/lib.rs:1427:9
  |
  = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant used
    --> src/lib.rs
     |
     |     assert!(<Dst as TransmuteFrom<Src>>::IS_TRANSMUTABLE);
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
